include/aws/common/task_scheduler.h

```diff
@@ -33,7 +33,12 @@ struct aws_task {
     struct aws_linked_list_node node;
     struct aws_priority_queue_node priority_queue_node;
     const char *type_tag;
-    size_t reserved;
+
+    /* honor the ABI compat */
+    union {
+        bool scheduled;
+        size_t reserved;
+    } abi_extension;
```

Comment:
This union could be anonymous and still achieve ABI compat.

Reply:
Yeah but it won’t compile because that’s a c12 feature apparently.